projects/2020-02-fenix-webrender/summary.Rmd

```diff
@@ -0,0 +1,272 @@
+The separate legs were delivered through the Play Store as different Fenix versions -- either 3.2.0 or 3.2.1.
+
+The randomization mechanisms sorting users into branches for each study were independent and our users were equally balanced
+between the optimizer-flag experiment branches, so I assert that the optimizer-flag study is ignorable and do not consider it further.
+
+```{r}
+summary %>%
+  count(branch, max_app_display_version) %>%
+  filter(n > 1) %>%
+  ggplot(aes(max_app_display_version, n, fill=branch, group=branch)) +
+    geom_col(position=position_dodge(w=0.75), width=0.7)
+```
+
+The plot below summarizes the difference in the median client's average performance metrics with Webrender enabled, with bootstrapped 95% confidence intervals.
+The "consistent `compositor_last_seen`" series is probably the correct value to reference.
+Because we don't completely understand why compositor was sometimes missing or inconsistent, the "all pings" series is presented for comparison,
+in case pings were missing for reasons that could bias the results, but
+(except for total checkerboarding time)
+the results only include pings where there was any activity for that metric.
+In cases where the "consistent compositor" and "all pings" results show different trends, we should interpret the results with caution.
+
+The "all pings" result for the total checkerboarding time was zero because the median client in the "all pings" case experienced no checkerboarding
+(and perhaps had no web content activity at all).
+
+```{r, fig.height=11, fig.width=4}
+bootstraps %>%
+  filter(term == "difference", only_consistent != "Inconsistent/null compositor_last_seen") %>%
+  mutate(metric=factor(
+    metric,
+    c("page_load_time_mean_s.safe_median", "checkerboard_potential_duration_sum_s.median", "composite_time_mean_ms.safe_median", "content_full_paint_time_mean_ms.safe_median", "content_paint_time_mean_ms.safe_median", "gfx_content_frame_time_from_vsync_mean.safe_median", "gfx_content_frame_time_from_paint_mean.safe_median"),
+    c("Mean page load time (s)", "Total checkerboarding duration (s)", "Mean composite time (ms)", "Mean content full paint time (ms)", "Mean content paint time (ms)", "Mean content frame time (% vsync)", "Mean content frame time (ms)")
+    )) %>%
+ggplot(aes(metric, statistic, ymin=conf.low, ymax=conf.high, color=only_consistent)) +
+    geom_pointrange(position=position_dodge(width=1)) +
+    geom_blank(aes(ymin=-conf.high, ymax=-conf.low)) +  # center everything at zero
+    coord_flip() +
+    geom_hline(yintercept=0) +
+    facet_wrap(~metric, scales="free", ncol=1) +
+    theme(axis.text.y=element_blank(), axis.ticks.y=element_blank(), panel.spacing.y=unit(2, "lines"), legend.position="top", legend.direction="vertical") +
+    labs(x="", title="Absolute difference for median\nuser") +
+    scale_color_discrete("Ping subset")
+```
+
+# Page load time
```

Comment:
Add a description for the plots below. (e.g., distributions of "consistent compositor_last_seen" pings for each metric). 
It would also be nice to add marker lines for the median, to add context to the plots above.